src/Roslyn.Suppressions/Roslyn.Suppressions/SupportedSuppressionBuilder.tt

```diff
@@ -115,24 +137,43 @@ namespace SonarLint.VisualStudio.Roslyn.Suppressions
                 .OrderBy(x => x);
         }
 
-        private IEnumerable<DiagnosticDescriptor> GetDiagnosticDescriptors(Type t)
+        private IEnumerable<string> GetIdsUsingReflection(Type t)
```

Comment:
Logically this code is the same as before; it just uses reflection instead of strong types, since at transformation the set of Roslyn assemblies being loaded is different and we get type mismatches (i.e. we have two different versions of `DiagnosticAnalyzer`, and we can't cast between them).

Risk-wise: there shouldn't really be any risk. We have unit tests that checks the generated code returns 400+ analyzer ids. If there is transformation error, the build will fail. If the transformation doesn't fail but we can't find any analyzers, the test will fail.